packages/dashboard-core-plugins/src/ChartPanelPlugin.tsx

```diff
@@ -64,53 +51,25 @@ async function createChartModel(
     }
   }
 
-  if (figureName == null && tableName == null) {
-    const figure = await fetchFigure();
+  if (tableName != null && tableName !== '') {
+    const table = (await panelFetch()) as DhType.Table;
+    new IrisGridUtils(dh).applyTableSettings(
+      table,
+      tableSettings,
+      getTimeZone(store.getState())
+    );
 
-    return ChartModelFactory.makeModel(dh, settings, figure);
+    return ChartModelFactory.makeModelFromSettings(dh, settings, table);
   }
 
-  if (figureName != null) {
-    let figure: DhType.plot.Figure;
-
-    if (metadata.type === dh.VariableType.FIGURE) {
-      const descriptor = {
-        ...metadata,
-        name: figureName,
-        type: dh.VariableType.FIGURE,
-      };
-      figure = await fetchObject<DhType.plot.Figure>(descriptor);
-    } else {
-      figure = await fetchFigure();
-    }
-
-    return ChartModelFactory.makeModel(dh, settings, figure);
-  }
+  const figure = (await panelFetch()) as DhType.plot.Figure;
 
-  const descriptor = {
-    ...metadata,
-    name: tableName,
-    type: dh.VariableType.TABLE,
-  };
-  const table = await fetchObject<DhType.Table>(descriptor);
-  new IrisGridUtils(dh).applyTableSettings(
-    table,
-    tableSettings,
-    getTimeZone(store.getState())
-  );
-
-  return ChartModelFactory.makeModelFromSettings(
-    dh,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    settings as any,
-    table
-  );
+  return ChartModelFactory.makeModel(dh, settings, figure);
 }
 
 export const ChartPanelPlugin = forwardRef(
   (props: WidgetPanelProps<DhType.plot.Figure>, ref: React.Ref<ChartPanel>) => {
-    const dh = useApi();
-    const fetchObject = useObjectFetcher();
+    const deferredApi = useContext(DeferredApiContext);
```

Comment:
This makes me wonder if we should still have `useApi`. Could `useApi` cause unknown bugs in DHE because we don't have the same complexity in DHC? I guess either way we would have to be cognizant of when to pass the metadata

Also, why not `useDeferredApi(metadata)` here?